app/components/school-competencies-expanded.js

```diff
@@ -47,8 +47,8 @@ export default Component.extend({
         this.set('bufferedCompetencies', competencies.toArray());
```

Comment:
@stopfstedt I think you are correct and this is the issue. It is going to get called a lot. And each invocation is going to start that promise chain and at some point they're going to come back and try and call the set. A temp fix could be to use `Ember.set` here which I believe protects against calling set on a destroyed object automatically, but if it doesn't then we could add a check like `if (this.isDestroyed || this.isDestroying) { return; }` here.

Reply:
none of these suggestions worked for me. attempting to rewrite this component and its sub-components further.